### ISSUE TITLE
Provide mechanism to permit status line to extend under sidebar

### ISSUE THREAD

**Issue by kinnison** [neomutt/neomutt#2903]
## What would you like to be able to do?

I would like to be able to make the status line extend underneath the sidebar when it is visible.

## What needs to change in NeoMutt?

I'm not sure what the config would be called, but perhaps `set status_over_sidebar=yes` might work

## Have you tried any alternative methods?

I have no idea how I'd alter the core rendering of the status line.

## Additional Info

Currently my status line looks like this:

![duff-statusline](https://user-images.githubusercontent.com/1469421/115309878-ca6ee980-a164-11eb-8f90-2eea623796e5.png)

I want to be able to configure neomutt to display like my similarly-configured Debian-based mutt-patched does:

![better-sidebar](https://user-images.githubusercontent.com/1469421/115309932-dd81b980-a164-11eb-8f4b-149fd1e91367.png)

**Comment by flatcap** [comment]
**tl;dr** Sorry, no.

Explaining why not takes a little effort, but I have some diagrams :-)

### History

Historically, the Mutt code used to write to absolute coordinates on the screen.
If you're horrified by this, you understand the problems NeoMutt is trying to fix ;-)
The original sidebar patch had to hack the layout code all over the place.

To simplify the code, Mutt introduced Windows to divide the screen into regions.
At the time I suggested that they should be nested, then Mutt and NeoMutt parted ways.
Mutt's code still has lots of hard-coded behaviour.
NeoMutt's code has a trivial reflow algorithm, which follows simple rules.

### Index/Sidebar

Here's what NeoMutt's Window layout looks like for the Index/Sidebar:
<a href="https://raw.githubusercontent.com/neomutt/gfx/master/screenshots/window/dlg-index-sidebar.svg"><img width="800" src="https://raw.githubusercontent.com/neomutt/gfx/master/screenshots/window/dlg-index-sidebar.svg"></a>

**Key**:
- <kbd>R</kbd> -- Root Window
- <kbd>A</kbd> -- All Dialogs
- <kbd>D</kbd> -- Dialog wrapper around the Index
- <kbd>C</kbd>, <kbd>I</kbd> -- Invisible containers

### Index/Pager/Sidebar

Now you might think, "well, rearrange the windows a bit and it'll be fine"
But then you open the Pager (with `$pager_index_lines` set).

Here's what NeoMutt's Window layout looks like for the Index/Pager/Sidebar:
<a href="https://github.com/neomutt/gfx/raw/master/screenshots/window/dlg-index-pager-sidebar.svg"><img width="800" src="https://github.com/neomutt/gfx/raw/master/screenshots/window/dlg-index-pager-sidebar.svg"></a>

The Status Bar, which is really the Index Bar, has moved.
Now we see that the it is tightly bound to the Index itself.

By nesting the Windows, NeoMutt creates better encapsulation of data and more independence in the code,
at the cost of a shorter status bar.
Sorry.

---

Lots more diagrams, here: https://github.com/neomutt/gfx/tree/master/screenshots

**Comment by kinnison** [comment]
That was a beautifully clear explanation, thank you @flatcap -- I wonder then if I might suggest the introduction of an (optional) sidebar status line?  I could close this issue and open a fresh one when I've thought through what it might be nice for it to contain if that would work for you?  I find the discontinuity of the line there to be exceedingly jarring, it makes it look like the sidebar and the command line / message window bleed into each other which is surprisingly distracting for me.

Would you be able to be more positive about the idea of a sidebar status line, or are there strong technical reasons to reject such an idea?

**Comment by flatcap** [comment]
> That was a beautifully clear explanation, thank you @flatcap

You're welcome

> I might suggest the introduction of an (optional) sidebar status line?
> if that would work for you?

Yes, I open to suggestions

> I find the discontinuity of the line there to be exceedingly jarring
> it makes it look like the sidebar and the command line / message window bleed into each other

Good point.
The Windows don't have edges, or colours, to differentiate them.
We can probably do something about _that_.

> I could close this issue and open a fresh one

Yes please, or you can find us on IRC:  `#neomutt` on `irc.freenode.net` ([web client](https://webchat.freenode.net/?channels=neomutt))

**Comment by kinnison** [comment]
I have opened #2905.
